### PR TITLE
Add cmake for newer version of libjpeg-turbo

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Newer version uses of libjpeg-turbo uses cmake.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py [hunk 1]
```diff
@@ -46,3 +47,19 @@ class LibjpegTurbo(AutotoolsPackage):
     # TODO: Implement the selection between two supported assemblers.
     # depends_on("yasm", type='build')
     depends_on("nasm", type='build')
+    depends_on('cmake', type='build', when="@1.5.90:")
+
+    @when('@:1.5.3')
+    def install(self, spec, prefix):
+        configure('--prefix=%s' % prefix)
+        make
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: These should be `make()`, it's a function, not a statement.

Comment (s-sajid-ali) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: Done.

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py [hunk 2]
```diff
@@ -25,15 +25,16 @@
...
 
 
-class LibjpegTurbo(AutotoolsPackage):
+class LibjpegTurbo(Package):
     """libjpeg-turbo is a fork of the original IJG libjpeg which uses SIMD to
        accelerate baseline JPEG compression and decompression. libjpeg is a
        library that implements JPEG image encoding, decoding and
        transcoding."""
+    # https://github.com/libjpeg-turbo/libjpeg-turbo/blob/master/BUILDING.md
+    homepage = "https://github.com/libjpeg-turbo/libjpeg-turbo"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: I would actually use https://libjpeg-turbo.org/ as the homepage. That's what the previous homepage redirects to.

Comment (s-sajid-ali) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: For some reason the building instructions for the 1.6.x was broken from there but that's fine.

**Pull request merged**